echo-scheduler/src/main/groovy/com/netflix/spinnaker/echo/config/SchedulerConfiguration.groovy

```diff
@@ -16,79 +16,113 @@
 
 package com.netflix.spinnaker.echo.config
 
-import com.netflix.fenzo.triggers.TriggerOperator
-import com.netflix.fenzo.triggers.persistence.InMemoryTriggerDao
-import com.netflix.fenzo.triggers.persistence.TriggerDao
-import com.netflix.scheduledactions.ActionsOperator
-import com.netflix.scheduledactions.DaoConfigurer
-import com.netflix.scheduledactions.persistence.ActionInstanceDao
-import com.netflix.scheduledactions.persistence.ExecutionDao
-import com.netflix.scheduledactions.persistence.InMemoryActionInstanceDao
-import com.netflix.scheduledactions.persistence.InMemoryExecutionDao
-import com.netflix.scheduledactions.web.controllers.ActionInstanceController
+import com.netflix.spinnaker.echo.scheduler.actions.pipeline.PipelineConfigsPollingJob
+import com.netflix.spinnaker.echo.scheduler.actions.pipeline.PipelineTriggerJob
+import com.netflix.spinnaker.echo.scheduler.actions.pipeline.TriggerListener
+import com.netflix.spinnaker.echo.scheduler.actions.pipeline.TriggerConverter
+import com.netflix.spinnaker.kork.sql.config.DefaultSqlConfiguration
 import com.squareup.okhttp.OkHttpClient
+import org.quartz.JobDetail
+import org.quartz.Trigger
+import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.beans.factory.annotation.Value
 import org.springframework.boot.autoconfigure.condition.ConditionalOnExpression
 import org.springframework.context.annotation.Bean
-import org.springframework.context.annotation.ComponentScan
 import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.Import
+import org.springframework.scheduling.quartz.JobDetailFactoryBean
+import org.springframework.scheduling.quartz.SchedulerFactoryBean
+import org.springframework.scheduling.quartz.SimpleTriggerFactoryBean
 import retrofit.client.Client
 import retrofit.client.OkClient
 
+import javax.sql.DataSource
 import java.util.concurrent.TimeUnit
 
 @Configuration
 @ConditionalOnExpression('${scheduler.enabled:false}')
-@ComponentScan(["com.netflix.spinnaker.echo.scheduler", "com.netflix.scheduledactions"])
+@Import(DefaultSqlConfiguration)
```

Comment:
Will scheduled triggers continue to work for users without a SQL backend configured?

Reply:
Yes, will work as it does today, with in-memory store. I will add full/validated SQL support in a subsequent PR

Reply:
Do we need to add support for in-memory solutions to detect misfires? With the in-memory topology today, we have the compensation job which looks back on all pipelines configured with cron and determines if any schedules were missed and will fire those immediately. Do we have a similar capability for quartz in-memory? SQL will handle this correctly, as the last fired timestamp is tracked centrally.

Reply:
I will bring back the compensation job!

Reply:
it is back!